RestApi/Sepes.RestApi/Controllers/PodController.cs

```diff
@@ -30,6 +31,37 @@ public async Task<Pod> createPod([FromBody] PodInput input)
         {
             return await _pod.CreateNewPod(input.podName, input.studyID);
         }
+
+        [HttpPost("nsg/create")]
+        public async Task createNsg([FromBody] NsgModel data)
+        {
+            await _pod.createNsg(data.securityGroupName, data.resourceGroupName);
+        }
+        [HttpPost("nsg/delete")]
+        public async Task deleteNsg([FromBody] NsgModel data)
+        {
+            await _pod.deleteNsg(data.securityGroupName, data.resourceGroupName);
+        }
+
+
+
+        [HttpPost("nsg/apply")]
+        public async Task applyNsg([FromBody] NsgModel data)
+        {
+            await _pod.applyNsg(data.resourceGroupName, data.securityGroupName, data.subnetName, data.networkName);
+        }
+
+        [HttpPost("nsg/remove")]
+        public async Task removeNsg([FromBody] NsgModel data)
+        {
+
+            await _pod.removeNsg(data.resourceGroupName, data.subnetName, data.networkName);
+        }
+        [HttpDelete("nsg/purgeunused")]
+        public async Task<UInt16> deleteUnused()
+        {
+            return await _pod.deleteUnused();
+        }
```

Comment:
Lets keep it out off the controller. If you want to test something you can create a DevController where you add temp and play-around APIes.